src/query/service/src/interpreters/interpreter_table_create.rs

```diff
@@ -115,7 +112,7 @@ impl Interpreter for CreateTableInterpreter {
                 }
             }
             None => {
-                if name_not_duplicate {
+                if !catalog.exists_table(tenant.as_str(), &self.plan.database, &self.plan.table) {
                     return Err(ErrorCode::UnknownTableEngine(format!(
```

Comment:
Actually I think this None check is useless.

In parser if engine not support we already return err

```
MySQL [(none)]> create table c(id int) engine=SDFS;
ERROR 1105 (HY000): Code: 1005, Text = error: 
  --> SQL:1:31
  |
1 | create table c(id int) engine=SDFS
  | ------                        ^^^^ expected `NULL`, `MEMORY`, `FUSE`, `VIEW`, or `RANDOM`
  | |                              
  | while parsing `CREATE TABLE [IF NOT EXISTS] [<database>.]<table> [<source>] [<table_options>]`

```

Reply:
I see.
We can remove this None check.

Reply:
If we create table:
```
create table t1(a int);
```

The default storage engine is set during planner?

Reply:
> If we create table:
> 
> ```
> create table t1(a int);
> ```
> 
> The default storage engine is set during planner?

In binder https://github.com/datafuselabs/databend/blob/95dd9dc7114070e82dfb5dab6fec8aa077cbe5cf/src/query/sql/src/planner/binder/ddl/table.rs#L332